packages/opencensus-web-core/src/trace/model/tracer.ts

```diff
@@ -22,62 +22,85 @@ import { randomTraceId } from '../../common/id-util';
 
 /** Tracer manages the current root span and trace header propagation. */
 export class Tracer extends TracerBase implements webTypes.Tracer {
+  private static singletonInstance: Tracer;
+
+  /** Gets the tracer instance. */
+  static get instance(): Tracer {
+    return this.singletonInstance || (this.singletonInstance = new this());
+  }
+
+  // Variable to store current root span in case the Zone global variable is not present.
+  // For that case we only need to store only one current root span.
+  private rootSpan = new RootSpan(this);
```

Comment:
Could we call this something like `rootSpanNoZone` or similar to indicate that it's only used if there is no Zone available?